R/utils.R

```diff
@@ -528,6 +528,18 @@ try_redcapr <- function(expr, call = caller_env()) {
   )
   error$call <- call
 
+  # List to store components of warning so we can look them up unambiguously
+  warning <- list()
+
+  # URI and token we want are in the env associated with expr
+  warning$redcap_uri <- env$redcap_uri
+  warning$token <- env$token
+
+  # Defaults for other error components
+  warning$message <- c("!" = "The {.pkg REDCapR} export operation produced an unexpected warning.")
+  warning$class <- "REDCapTidieR_cond"
+  warning$call <- call
+
```

Comment:
I actually don't think we need to duplicate this from L518-L529 since most of the defaults are the same. I would change all the `error$` to `condition$` just so the naming is consistent and then have the `cli_warn()` call on L567 be:

```r
cli_warn(
  c("!" = "What Stephan suggested ;)"),
  call = condition$call,
  parent = cnd,
  class = c("unexpected_warning", condition$class)
)
```

Reply:
That's fine with me, I chose this on the off chance that down the line we want to throw more specific warnings and increase the list options. But I like simplicity better.